### PR TITLE
rust: Followup changes for multi-account support (#93)

### PR DESCRIPTION
**Pull request by shesek**
Expose the new functionality through the CPP interface, with some adjustments to match the `ga_auth_handlers` flow.

### CODE REVIEW

**Comment by LeoComandini** [comment]
In `subprojects/gdk_rust/src/interface.rs:92`, `s/"pointer": 0,/"pointer": subaccount.account_num,/`, otherwise pointer is useless in get_subaccount(s).

**Comment by LeoComandini** [comment]
`get_subaccounts` returns the account type lowercase (`p2pkh`), but `create_subaccount` expects type in CamelCase (`P2pkh`). Can we make that always lowercase?

File: subprojects/gdk_rust/gdk_common/src/model.rs [hunk 1]
```diff
@@ -148,7 +148,12 @@ pub struct GetAddressOpt {
 
 #[derive(Serialize, Deserialize, Debug, Clone)]
 pub struct CreateAccountOpt {
+    pub subaccount: u32,
     pub name: String,
+}
```
Comment (LeoComandini) [inline subprojects/gdk_rust/gdk_common/src/model.rs:153]: Integration tests needs updating, `CreateAccountOpt` does not have `script_type` field.

Comment (shesek) [reply subprojects/gdk_rust/gdk_common/src/model.rs:153]: Addressed by https://github.com/Blockstream/gdk/pull/102/commits/cb8d1b1eb6e8c16cd3397816e2d91c21f4c67ef7.

**Comment by LeoComandini** [comment]
ACK cb8d1b1eb6e8c16cd3397816e2d91c21f4c67ef7, code review, tested locally, gitlab CI passed

**Pull request merged**